lib/ex_doc/formatter/html/templates.ex

```diff
@@ -53,9 +53,10 @@ defmodule ExDoc.Formatter.HTML.Templates do
 
   def synopsis(doc) when is_binary(doc) do
     doc =
-      case :binary.split(doc, "</p>") do
-        [left, _] -> String.trim_trailing(left, ":") <> "</p>"
-        [all] -> all
+      Regex.run(~r|<p>((?:(?!</p>).)*?):*</p>|, doc)
+      |> case do
+        nil -> doc
+        [_, first_paragraph_text] -> "<p>" <> first_paragraph_text <> "</p>"
       end
```

Comment:
We are kind of assuming that it starts with <p>, so perhaps I would enforce that instead?

```
case String.trim_leading(doc) do
  "<p>" <> _ = doc ->
    doc =
      case :binary.split(doc, "</p>") do
        [left, _] -> String.trim_trailing(left, ":") <> "</p>"
        [all] -> all
      end

    Regex.replace(~r|(<[^>]*) id="[^"]*"([^>]*>)|, doc, ~S"\1\2", [])

  _ ->
    nil
end
```

Alternatively we capture the synopsis over the AST instead of the rendered document. This would also make it easier to remove links and what not.